### PR TITLE
ADCM-1446 Update config functions

### PR DESCRIPTION
**Pull request by a-alferov**

### CODE REVIEW

File: adcm_client/objects.py [hunk 1]
```diff
@@ -296,29 +272,30 @@ def config(self, full=False):
 
     @allure_step("Save config")
     def config_set(self, data):
+        config = self.config(full=True)
```
Comment (acmnu) [inline adcm_client/objects.py:275]: No needs for config read here. It is not used in "then" part of "if" in line 276.

File: adcm_client/objects.py [hunk 2]
```diff
@@ -296,29 +272,30 @@ def config(self, full=False):
...
-    def config_set_diff(self, data):
-        config = self.config()
-        for gk, gv in data.items():
-            if isinstance(gv, dict):
-                for k, v in gv.items():
-                    config[gk][k] = v
-            else:
-                config[gk] = gv
-        self.config_set(config)
+    def config_set_diff(self, data, full=False):
```
Comment (acmnu) [inline adcm_client/objects.py:287]: Why do you need to pass full explicit? You could find out about from data structure as your done in config config_set function.

File: adcm_client/objects.py [hunk 3]
```diff
@@ -296,29 +272,30 @@ def config(self, full=False):
...
+
+        def update(d, u):
```
Comment (acmnu) [inline adcm_client/objects.py:289]: I think, that you need to care about attrs too. If we have a function that allow us to pass config_diff, why we doesn't allow the same for attrs?

File: adcm_client/objects.py [hunk 4]
```diff
@@ -296,29 +272,30 @@ def config(self, full=False):
...
+            for key, value in u.items():
+                if isinstance(value, abc.Mapping):
+                    d[key] = update(d.get(key, {}), value)
```
Comment (acmnu) [inline adcm_client/objects.py:292]: I believe that you could use dict.update(another_dict) function as part of this merge procedure. See https://docs.python.org/3/library/stdtypes.html#dict.update

**Review by acmnu — CHANGES_REQUESTED** [review]

File: adcm_client/objects.py (hunk 4 above)
Comment (a-alferov) [reply adcm_client/objects.py:292]: I think that we need a deep update, and the built-in method cannot do this.

File: adcm_client/objects.py (hunk 2 above)
Comment (a-alferov) [reply adcm_client/objects.py:287]: I find it a bad idea to explicitly check for the presence of keys in a structure

File: adcm_client/objects.py (hunk 1 above)
Comment (a-alferov) [reply adcm_client/objects.py:275]: Yes you are right.

File: adcm_client/objects.py (hunk 3 above)
Comment (a-alferov) [reply adcm_client/objects.py:289]: What do you suggest? Add a new method `attr_set_diff()` ?

**Review by acmnu — APPROVED** [review]

File: adcm_client/objects.py [hunk 5]
```diff
@@ -301,29 +277,31 @@ def config(self, full=False):
...
             if data["attr"] is None:
-                data["attr"] = []
-            history_entry = self._subcall(
-                'config', 'history', 'create',
-                config=data["config"],
-                attr=data["attr"],
-            )
-            return history_entry
+                data["attr"] = {}
+            history_entry = self._subcall('config', 'history', 'create', **data)
```
Comment (skhomuti) [inline adcm_client/objects.py:285]: these changes resulted in an error
`{ParameterError}{'date': 'Unknown parameter.', 'id': 'Unknown parameter.'}`
`config(full=True)` returns object with extra fields that are not needed to update the config

**Review by skhomuti — APPROVED** [review]

**Pull request merged**